### ISSUE TITLE
0.7.8 gem checksum mismatch

### ISSUE THREAD

**Issue by tsigo** [Azure/azure-sdk-for-ruby#755]
The SHA256 sum on https://rubygems.org/gems/azure/versions/0.7.8 is `6210f164e1ece6b68961a68cfc7a16d764033d53aa3016a6e51836a383a25798` but there's a mismatch with the downloaded gem:

```
% shasum -a 256 azure-0.7.8.gem
b5b99af4ac7153e074742a55ed1e3d1d9522ad6d79d94edba42ea172f5c22198  azure-0.7.8.gem
```

And Bundler is reporting the same issue:

```
Bundler cannot continue installing azure (0.7.8).
The checksum for the downloaded `azure-0.7.8.gem` does not match the checksum given by the server. This means the contents of the downloaded gem is different from what was uploaded to the server, and could be a potential security issue.

To resolve this issue:
1. delete the downloaded gem located at: `/Users/tsigo/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/azure-0.7.8/azure-0.7.8.gem`
2. run `bundle install`

If you wish to continue installing the downloaded gem, and are certain it does not pose a security issue despite the mismatching checksum, do the following:
1. run `bundle config disable.checksum_validation true` to turn off checksum verification
2. run `bundle install`

(More info: The expected SHA256 checksum was "6210f164e1ece6b68961a68cfc7a16d764033d53aa3016a6e51836a383a25798", but the checksum for the downloaded gem was "b5b99af4ac7153e074742a55ed1e3d1d9522ad6d79d94edba42ea172f5c22198".)
```

**Comment by sarangan12** [comment]
@tsigo 

Thanks for reporting the issue. This looks like an issue related to rubygems. I have created an issue with them. Once I get more updates, I will add them here.

**Comment by sarangan12** [comment]
Context:

We are using the Travis CI to publish the gem to rubygems. This instructions to publish the gems are provided in the [.travis.yml](https://github.com/Azure/azure-sdk-for-ruby/blob/asm/.travis.yml#L11).

The problem is that we have mentioned 4 versions of ruby in the file. For the deploy operation, all the 4 versions run in parallel and try to publish the gem 4 times which causes a race condition. As a result, the S3 (which stores the actual gem), accepts the first one and rejects all the others. While the module to store the checksum values probably accepts the last one. 

In order to fix the issue, we have to mention the version (just one) to run the deploy operation. Refer [#1551](https://github.com/rubygems/rubygems.org/issues/1551) for more context.

Note, On April 17, 2017 two gems were released (azure & azure-core). The azure-core gem does not have this issue and so we need not do anything there. But the azure gem has a mismatch. In order to fix this issue, 0.7.9 version (exact copy on 0.7.8 is released)

**Comment by sarangan12** [comment]
@tsigo 
A new version has been released and the checksum validation is passing now. I am closing this issue now.
